lib/action.ml

```diff
@@ -10,6 +10,55 @@ let action_error msg = raise (Action_error msg)
 let log = Log.from "action"
 
 module Action (Github_api : Api.Github) (Slack_api : Api.Slack) = struct
+  let email_canonical_regex = Re2.create_exn {|\+[^]]*|@[^]]*|} (* Match email domain and everything after '+' *)
+
+  let username_ignored_chars = [ '.'; '-' ]
+  let username_to_slack_id_tbl = Stringtbl.empty ()
+
+  let canonicalize_username username =
```

Comment:
This function is simpler with a regex substitution:
```
let ignored_chars_regex = Re2.create_exn {|[\.\-]|}

let canonicalize_username s = String.lowercase_ascii @@ Re2.rewrite_exn ~template:"" ignored_chars_regex s
```

Reply:
I can also put this with `email_canonical_regex` for a single regex: `[\.\-]|\+[^]]*|@[^]]*` that would match domain, plus addressing, and `.` and `+`. Or do you think keeping them separate is better for readability?